packages/nvidia_nat_adk/src/nat/plugins/adk/tool_wrapper.py

```
@@ -0,0 +1,160 @@
+            # Set signature only if input_schema is provided
+            params: list[inspect.Parameter] = []
+            if input_schema is not None:
+                annotations = getattr(input_schema, "__annotations__", {}) or {}
+                for param_name, param_annotation in annotations.items():
+                    params.append(
+                        inspect.Parameter(
+                            param_name,
+                            inspect.Parameter.POSITIONAL_OR_KEYWORD,
+                            annotation=resolve_type(param_annotation),
+                        ))
+            setattr(func_to_wrap, "__signature__", inspect.Signature(parameters=params))
```

Ruff (0.13.1)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)